platform/mbed_error.c

```diff
@@ -45,6 +45,11 @@
                             }                                                       \
                         }
 
+#ifndef NDEBUG
+#define ERROR_REPORT    print_error_report
+#else
+#define ERROR_REPORT
+#endif
```

Comment:
Shall we not add an additional macro and instead just update the code as below:
```
#ifdef NDEBUG
#define print_error_report
#endif
```

Reply:
That would be confusing and may affect the readability to have actual function name to be nulled out. The macro bound to NDEBUG or not name clearly indicates the issue being addressed. Also its not a public macro, should not be a concern.

Reply:
@kjbracey-arm - Ok, I can change it to include args. And I would go with second version, I have found the first version to be confusing and affects readability.